src/levanter/data/text.py

```diff
@@ -522,7 +523,7 @@ def open_shard_at_row(self, shard_name: str, row: int) -> Iterator[str]:
             shard = dataset
 
         idx = 0
-        for _, doc in shard:
+        for _, doc in safe_enumerate_iterable(shard):
```

Comment:
i think i don't like this.  is this to suppress the http error? If so, and those failures aren't deterministic, then this breaks reproducibility, which is important for this project

 i'd rather crash if the underlying data source is broken.